tests/test_cyto_utils/test_collate.py

```diff
@@ -0,0 +1,172 @@
+import os
+import pathlib
+import pytest
+
+import pandas as pd
+from pycytominer.cyto_utils.collate import collate
+
+# Set constants
+BATCH = "2021_04_20_Target2"
+PLATE = "BR00121431"
+TEST_DIR = pathlib.Path(__file__).parents[1].absolute()
+ROOT_DIR = TEST_DIR.parent.absolute()
+TEST_CONFIG_LOCATION = ROOT_DIR.joinpath(
+    "pycytominer", "cyto_utils", "database_config", "ingest_config.ini"
+)
+TEST_DATA_LOCATION = TEST_DIR / "test_data" / "collate"
+TEST_BACKEND_LOCATION = (
+    TEST_DATA_LOCATION / "backend" / BATCH / PLATE / f"{PLATE}.sqlite"
+)
+TEST_CSV_LOCATION = TEST_DATA_LOCATION / "backend" / BATCH / PLATE / f"{PLATE}.csv"
+MAIN_CSV_LOCATION = TEST_DATA_LOCATION / "backend" / BATCH / PLATE / f"{PLATE}_main.csv"
+
+
+def cleanup():
+    if os.path.exists(TEST_BACKEND_LOCATION):
+        os.remove(TEST_BACKEND_LOCATION)
+
+    if os.path.exists(TEST_CSV_LOCATION):
+        os.remove(TEST_CSV_LOCATION)
+
+
+def validate(test_file, main_file, should_be_equal=True):
+    test = pd.read_csv(test_file)
+    main = pd.read_csv(main_file)
+    if should_be_equal:
+        pd.testing.assert_frame_equal(test, main)
+    else:
+        with pytest.raises(AssertionError):
+            pd.testing.assert_frame_equal(test, main)
+    return test, main
+
+
+def test_base_case():
+    cleanup()
+
+    collate(
```

Comment:
@kenibrewer - apologies for not seeing this sooner, but I wanted to raise an issue I noticed which seems to pertain to the changes with `collate.py` and how it is used by `collate_cmd.py`. In attempting to replicate this test using the command line through the poetry environment I'm seeing an error. I tested this same command with the master branch and things seem to work without error. Apologies in advance if I've somehow confused how this is implemented.

- **command**: (from within the pycytominer dir) `poetry run python pycytominer/cyto_utils/collate_cmd.py "2021_04_20_Target2" "pycytominer/cyto_utils/database_config/ingest_config.ini" "BR00121431" --base pycytominer/tests/test_data/collate --tmp-dir pycytominer/tests/test_data/collate --dont-add-image-features --dont-print`
- **error**: `sqlite3.OperationalError: no such table: main.Image`. When I inspected the SQLite output I found it was a seemingly 0 table file. Maybe this means cytominer-database was unable to complete the ingest?

Reply:
@d33bs There error you are experiencing is due to tests being moved from `/pycytominer/tests` to `/tests` . If you run the command below everything works as expected.
```
poetry run python pycytominer/cyto_utils/collate_cmd.py "2021_04_20_Target2" "pycytominer/cyto_utils/database_config/ingest_config.ini" "BR00121431" --base tests/test_data/collate --tmp-dir tests/test_data/collate --dont-add-image-features --dont-print
```

In the process of researching this possible bug, I did discover an [important detail](https://docs.python.org/3/library/sqlite3.html#how-to-use-the-connection-context-manager) about the sqlite3 connection context manager. Specifically, it does not automatically close connections after it leaves the with statement. Therefore, I added an explicit `connection.close()` to the code.

Reply:
Thanks and great find! I tested your command and things worked properly on my end.